tpp-app/src/services/cycle/CycleService.js

```diff
@@ -69,25 +72,30 @@ export {getLastPeriodsEnd};
 
 async function isOnPeriod (calendar = null) {
 
-  let current = await this.GETMostRecentPeriodStartDate(calendar) // most recent period start date
+  try {
+    let current = await this.GETMostRecentPeriodStartDate(calendar) // most recent period start date
 
-  var yesterday = subDays(current, 1);
+    var yesterday = subDays(current, 1);
```

Comment:
There's a lot of instances of `var` tbh LOL. At this point it might take a big Ctrl+F whole repo. It's not a big deal though as long as it's not breaking anything.

Reply:
made changes! except for notification files since Edward is still working on those